Sources/PenguinStructures/ArrayStorage.swift

```diff
@@ -122,7 +162,58 @@ extension ArrayStorageImplementation {
     }
     return r
   }
+  
+  /// Returns new storage having at least the given capacity, calling
+  /// `initialize` with pointers to the base addresses of self and the new
+  /// storage.
+  public func replacementStorage(
+    count newCount: Int,
+    minimumCapacity: Int,
+    initialize: (
+      _ selfBase: UnsafeMutablePointer<Element>,
+      _ replacementBase: UnsafeMutablePointer<Element>) -> Void
+  ) -> Self {
+    assert(minimumCapacity >= newCount)
+    let r = Self.create(minimumCapacity: minimumCapacity)
+    r.count = newCount
+    
+    withUnsafeMutableBufferPointer { src in
+      r.withUnsafeMutableBufferPointer { dst in
+        initialize(
+          src.baseAddress.unsafelyUnwrapped,
+          dst.baseAddress.unsafelyUnwrapped)
+      }
+    }
+    return r
+  }
 
+  /// Returns a copy of `self` after appending `x`, moving elements from the
+  /// existing storage iff `moveElements` is true.
+  ///
+  /// - Postcondition: if `count == capacity` on invocation, the result's
+  ///   `capacity` is `self.capacity` scaled up by a constant factor.
+  ///   Otherwise, it is the same as `self.capacity`.
+  /// - Postcondition: if `moveElements` is `true`, `self.count == 0`
+  ///
+  /// - Complexity: O(N).
+  @inline(never)
```

Comment:
Could you add a comment explaining why this is `@inline(never)`?

Reply:
Done